lib/puppet/transaction/event.rb

```diff
@@ -58,11 +60,13 @@ def to_data_hash
       'name' => @name,
       'status' => @status,
       'time' => @time.iso8601(9),
-      'redacted' => @redacted
+      'redacted' => @redacted,
+      'corrective_change' => @corrective_change,
     }
   end
 
   def property=(prop)
+    @property_instance = prop
```

Comment:
Be interesting to know, I was careful to have this avoid the YAML serialization at least. Its very easy to accidentally drop crap into the report for example without realizing it :-).